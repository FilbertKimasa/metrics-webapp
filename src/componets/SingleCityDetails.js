import React from 'react';

function SingleCityDetails() {
  return <div>t</div>;
}

export default SingleCityDetails;
