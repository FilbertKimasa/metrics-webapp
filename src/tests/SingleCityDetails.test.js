import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import SingleCityDetails from '../componets/SingleCityDetails';

const mockStore = configureMockStore();

describe('SingleCityDetails', () => {
  const initialState = {
    cities: {
      citiesData: [
        {
          id: '1',
          city: 'City 1',
          data: {
            list: [
              {
                components: {
                  Component1: 10,
                  Component2: 20,
                },
              },
            ],
          },
        },
      ],
    },
  };

  it('renders the component with correct city name and components', () => {
    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <SingleCityDetails />
        </BrowserRouter>
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
