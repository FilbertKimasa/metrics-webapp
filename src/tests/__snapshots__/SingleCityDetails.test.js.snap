// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleCityDetails renders the component with correct city name and components 1`] = `
<DocumentFragment>
  <div
    class="details-container"
  >
    <header
      class="details-header"
    >
      <a
        href="/"
      >
        <i
          class="material-icons back-arrow"
        >
          chevron_left
        </i>
      </a>
      <h2
        class="city-name"
      />
    </header>
  </div>
</DocumentFragment>
`;
